client/src/ui/diagram.html

```diff
@@ -78,143 +74,101 @@ <h5 class="modal-title" id="modal-title">Modal title</h5>
-
-            var $row = $(this),
-                name = $row.find("column1").text(),
+        let samlRequest = `{SAML_REQUEST}`;
+        let samlResponse = `{SAML_RESPONSE}`;
+        let oidcAuthzRequest = `{OIDC_AUTHZ_REQUEST}`;
+        let oidcAuthzResponse = `{OIDC_AUTHZ_RESPONSE}`;
+        let oidcTokenRequest = `{OIDC_TOKEN_REQUEST}`;
+        let oidcTokenResponse = `{OIDC_TOKEN_RESPONSE}`;
+
+        $xml.find("InboundAuthenticationConfig")
+            .find("InboundAuthenticationRequestConfigs")
+            .find("InboundAuthenticationRequestConfig").each(function () {
+
+            let $row = $(this),
                 job = $row.find("InboundAuthType").text();
 
+            if (JOB_TYPE_SAML_SSO == job) {
 
-            if (job == "samlsso") {
-                x.innerHTML += `<div class="card" id="samlsso-card" style="margin: 20px">
+                inboundAuthElem.innerHTML += `
+                        <div class="card" id="samlsso-card" style="margin: 20px">
                            <div class="text-center card-header-pills">
                                <div><b>` + job + `</b></div>
                             </div>
                         </div>`;
 
-                addConfig(samlRequest,samlResponse);
+                addSAMLSSOConfig(samlRequest, samlResponse);
+            } else if (JOB_TYPE_SAML_OPENID == job) {
+                inboundAuthElem.innerHTML += `
+                    <div class="card" style="margin: 20px" id="oidc-card">
+                        <div class="card" id="oidc-label">
+                            <b>` + job + `</b>
+                        </div>
+                        <div class="card-body" style="padding: 10px">
+                            <div class="card" style="margin-bottom: 5px; padding: 10px">
+                                <div class="text-center">
+                                    <h6>Authz Endpoint</h6>
+                                </div>
+                                <div style="padding-bottom: 10px"><button disabled id="authz-req-btn" class="btn btn-secondary btn-xs ">Request</button></div>
+                                <div><button disabled id="authz-res-btn" class="btn btn-secondary btn-xs ">Response</button></div>
+                            </div>
+                            <div class="card" style="padding: 10px">
+                                <div class="text-center">
+                                    <h6>Token Endpoint</h6>
+                                </div>
+                                <div style="padding-bottom: 10px"><button disabled id="token-req-btn" class="btn btn-secondary btn-xs ">Request</button></div>
+                                <div><button disabled id="token-res-btn" class="btn btn-secondary btn-xs ">Response</button></div>
+                            </div>
+                        </div>
+                    </div>
+                        `;
```

Comment:
If possible, move the templates to a separate file as constants and use them here to reduce clutter.

Reply:
It seems hard since adding another file need big code change.